Kentor.AuthServices.Tests/Configuration/SPOptionsTests.cs

```diff
@@ -96,5 +98,342 @@ public void SPOptions_MetadataCacheDuration_DefaultValue()
+        public void SPOptions_DecryptionCertificate_UseBothWhenFutureUnspecifiedPublished()
+        {
+            var subject = new SPOptions();
+            subject.ServiceCertificates.Add(new ServiceCertificate { Certificate = SignedXmlHelper.TestCert2 });
+            subject.ServiceCertificates.Add(new ServiceCertificate { Status = CertificateStatus.Future, Certificate = SignedXmlHelper.TestCert });
+
+            subject.DecryptionServiceCertificates.Count.Should().Be(2);
+            subject.DecryptionServiceCertificates[0].SerialNumber.Should().NotBe(subject.DecryptionServiceCertificates[1].SerialNumber);
+        }
+
+        [TestMethod]
+        public void SPOptions_DecryptionCertificate_UseBothWhenFutureEncryptionPublished()
+        {
+            var subject = new SPOptions();
+            subject.ServiceCertificates.Add(new ServiceCertificate { Certificate = SignedXmlHelper.TestCert2 });
+            subject.ServiceCertificates.Add(new ServiceCertificate { Use = CertificateUse.Encryption, Status = CertificateStatus.Future, Certificate = SignedXmlHelper.TestCert });
+
+            subject.DecryptionServiceCertificates.Count.Should().Be(2);
+            subject.DecryptionServiceCertificates[0].SerialNumber.Should().NotBe(subject.DecryptionServiceCertificates[1].SerialNumber);
+        }
+
+        [TestMethod]
+        public void SPOptions_DecryptionCertificate_EmptyWhenOnlySigning()
+        {
+            var subject = new SPOptions();
+            subject.ServiceCertificates.Add(new ServiceCertificate { Use = CertificateUse.Signing, Certificate = SignedXmlHelper.TestCert2 });
+
+            subject.DecryptionServiceCertificates.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void SPOptions_MetadataCertificates_EmptyWhenNoneAdded()
+        {
+            var subject = new SPOptions();
+
+            subject.MetadataCertificates.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void SPOptions_MetadataCertificates_Single()
+        {
+            var subject = new SPOptions();
+            subject.ServiceCertificates.Add(new ServiceCertificate { Certificate = SignedXmlHelper.TestCert });
+
+            var result = subject.MetadataCertificates;
+            result.Count.Should().Be(1);
+            result[0].Certificate.SerialNumber.Should().Be(SignedXmlHelper.TestCert.SerialNumber);
+        }
+
+        [TestMethod]
+        public void SPOptions_MetadataCertificates_BothBecomesSigningWhenFuturePublished()
+        {
+            var subject = new SPOptions();
+            subject.ServiceCertificates.Add(new ServiceCertificate { Certificate = SignedXmlHelper.TestCert });
+            subject.ServiceCertificates.Add(new ServiceCertificate { Status = CertificateStatus.Future, Certificate = SignedXmlHelper.TestCert2 });
+
+            var result = subject.MetadataCertificates;
+            result.Count.Should().Be(2);
+            result[0].Use.Should().NotBe(result[1].Use);
```

Comment:
Now I see. I read "Both" in the test title as "Both Certificates", not as "Use=Both". Might be worth a short comment to avoid confusing someone else reading the code later.